### ISSUE TITLE
Visibility Widget does not return a value/error when point selected is outside accepted area

### ISSUE THREAD

**Issue by ACueva** [Esri/solutions-webappbuilder-widgets#799]
### Widget
Visibility
### Version of widget
2.2
### Bug or Enhancement
Bug
### Repo Steps or Enhancement details

1. Click on Visibility Widget.
2. Select a point outside your area of interest.
3. Observe the spinning mouse cursor.

Behind the scene the Viewshed service has already kicked back an error message.

```
{"error":{"code":400,"message":"Unable to complete operation.","details":["Error executing tool. Viewshed Job ID: j277631ec1445450db7bd8dd6e2dcb0bf"]}}
```

This can be very frustrating as a user since at this point no other widgets are working.  Only option at this point is to refresh the page, possibly losing all the other work you've done.

**Comment by ACueva** [comment]
@nfeuerstein IMO, this might be a showstopper.

**Comment by nfeuerstein** [comment]
@ACueva what would be the expected experience?  The user clicks on a point of the map outside the area of interest and...warning?

**Comment by ACueva** [comment]
@nfeuerstein Not sure to be honest.  At the very least I would expect the widget to handle the error and allow the user to continue working.  Right now if you hit an error it basically locks you out of doing anything until you refresh the browser.

**Comment by adgiles** [comment]
@ACueva 

Added a handler to the widget to handle an error being returned from the gp service

![untitled1](https://cloud.githubusercontent.com/assets/3740334/22062869/da87fb22-dd73-11e6-8069-2d4e6aee9684.png).

What I have found is that the error can take a while to return depending on the file size of the elevation raster being used within the service.

My guess is that the arcpy.Clip_management function within the python is being run against the whole extent of the elevation raster file (which could be very large) if the point is outside the raster extent. 

I would recommend that the python script is amended to check that the point (or the extent of the buffered point) falls with the extent of the elevation raster, and returns an error if not, at the beginning of the script before the clip is attempted.

@mfunk Is this possible?

**Comment by kgonzago** [comment]
@mfunk @adgiles Suggestion: We add another tool to the toolbox which passes back the extent of the raster. This new operation will allow retrieval of the raster extent upon launching of the widget and perform the "is point within extent" check client side. User will immediately know if the point of interest either falls within or outside the extent of the raster without the need for waiting.

**Comment by adgiles** [comment]
@kgonzago 

Not sure of the need a separate tool, shouldn't we be able to use arcpy.raster 

http://resources.arcgis.com/en/help/main/10.1/index.html#//018z00000051000000

to get the extent of the raster, then create a polygon from the extent then use the geometry functions to determine if the point is within, contains etc..

http://resources.arcgis.com/en/help/main/10.1/index.html#//018z00000070000000

**Comment by kgonzago** [comment]
@adgiles Agreed on ease of GP service set up. Given our tight timeline, let's proceed with folding this simple check into the script. 

@mfunk Will you be able to assist?

**Comment by nfeuerstein** [comment]
Moving GP service enhancement to version after-next.

**Comment by nfeuerstein** [comment]
Verify gigzy's fix produces error and close out if it does.  A new issue should be created for the enhancement to have client-side checks.

**Comment by mfunk** [comment]
Working on a check in the viewshed.py to make sure point falls within surface **extents** and throws exception if not.

In trying to keep the solution 'light' and not expend extra processing time I'm working with the surface extents and not it's data "mask" area. If we need that level of check it will add more time.

**Comment by mfunk** [comment]
@ACueva 
Fixed in PR #803.
test service is updated, please retest.

**Comment by adgiles** [comment]
Confirmed fix works for both single raster and mosaic dataset
